nectar/downloaders/threaded.py

```diff
@@ -88,15 +88,16 @@ def __init__(self, config, event_listener=None, tries=DEFAULT_TRIES, session=Non
         # set of locations that produced a connection error
         self.failed_netlocs = set([])
 
-        self.session = session or build_session(config)
+        if not session:
```

Comment:
I made this change since if you choose to hand the session to ``__init__`` it'll toss out any adapter configuration, which is very unexpected. It's not directly related to #2111.